app/page/send/index.tsx

```diff
@@ -149,26 +152,26 @@ const SendAmount = ({
   const { token, txTo } = txInfo
 
   const { provider } = useContext(ProviderContext)
-  const network = useNetwork()
 
   const [inputAmount, setInputAmount] = useState<string>("0")
+  const [transferAmount, setTransferAmount] = useState(0n)
```

Comment:
It seems that `transferAmount` is a computed value from `inputAmount`.
How about we calculate `transferAmount` in `handleSend`?
```
const handleSend = () => {
  const transferAmount = ethers.parseUnits(inputAmount, token.decimals)
  ...
}
```

Reply:
I actually encountered rounding problem yesterday. If we click `MAX` in the app, the input amount would be rounded up to at most 2 decimals, and computing it back later might have over balance problem.

For example, suppose the account has USDT (decimals = 6) balance `56718` (0.56718), the input amount would be formatted to `0.57`. When user is about to send, computing from the input amount would get a transfer amount `57000`, which exceeds the user's actual balance.

Reply:
Glad found this issue in this time :rainbow:, not during the demo.

Reply:
Um.. I see 🧐

Maybe we need a function to handle max amount for sending token (like [metamask](https://github.com/MetaMask/metamask-extension/blob/6eafd291ab1bb942d421ea0f2deca7a791efa381/ui/ducks/send/send.js#L1079) or do some workaround like rounding down) in the future.